plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/KafkaEventPublisher.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.eventlistener.kafka;
+
+import io.airlift.log.Logger;
+import io.trino.plugin.eventlistener.kafka.metadata.EnvMetadataProvider;
+import io.trino.plugin.eventlistener.kafka.metadata.MetadataProvider;
+import io.trino.plugin.eventlistener.kafka.metadata.NoOpMetadataProvider;
+import io.trino.plugin.eventlistener.kafka.metrics.KafkaEventListenerJmxStats;
+import io.trino.plugin.eventlistener.kafka.producer.KafkaProducerFactory;
+import io.trino.plugin.eventlistener.kafka.producer.SSLKafkaProducerFactory;
+import io.trino.spi.eventlistener.QueryCompletedEvent;
+import io.trino.spi.eventlistener.QueryCreatedEvent;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class KafkaEventPublisher
+{
+    private static final Logger LOG = Logger.get(KafkaEventPublisher.class);
+
+    private final KafkaProducer<String, GenericRecord> kafkaProducer;
+    private final KafkaRecordBuilder kafkaRecordBuilder;
+    private final KafkaEventListenerJmxStats stats;
+
+    KafkaEventPublisher(KafkaEventListenerConfig config, KafkaProducerFactory producerFactory, KafkaEventListenerJmxStats stats)
+            throws Exception
+    {
+        this.stats = stats;
+
+        String startedTopic = config.getCreatedTopicName().orElse("");
+        String completedTopic = config.getCompletedTopicName().orElse("");
+
+        Map<String, String> configOverrides = config.getKafkaClientOverrides();
+        LOG.info("Kafka Event Listener: Creating Kafka publisher (SSL=%s) for topics: %s/%s with excluded fields: %s and kafka config overrides: %s",
+                producerFactory instanceof SSLKafkaProducerFactory, startedTopic, completedTopic, config.getExcludedFields(), configOverrides);
+        kafkaProducer = producerFactory.producer(configOverrides);
+        checkConnectivityToBrokers(config.getPublishCreatedEvent() ? startedTopic : completedTopic, config.getRequestTimeout());
```

Comment:
prepend with `if (config.getPublishCreatedEvent() || config.getPublishCompletedEvent()) {...`

I think scenario when both are off is valid

Reply:
Added this check to one layer above, to `KafkaEventListener`